### ISSUE TITLE
Resolve "flake8 does not generate error when given a non-existent file on the command line" - [merged]

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#878]
In GitLab by @shukron on Feb 21, 2018, 08:22

_Merges 405-nonexisting-files -> master_

Closes #405

**Comment by asottile** [comment]
In GitLab by @codecov on Feb 21, 2018, 08:22

# [Codecov](https://codecov.io/gl/pycqa/flake8/pull/227?src=pr&el=h1) Report
> Merging [#227](https://codecov.io/gl/pycqa/flake8/pull/227?src=pr&el=desc) into [master](https://codecov.io/gl/pycqa/flake8/commit/f8344997267b8ca87a96c690a3515a443005b653?src=pr&el=desc) will **increase** coverage by `0.41%`.
> The diff coverage is `100%`.

[![Impacted file tree graph](https://codecov.io/gl/pycqa/flake8/pull/227/graphs/tree.svg?width=650&height=150&src=pr&token=QL6TVnaXih)](https://codecov.io/gl/pycqa/flake8/pull/227?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master     #227      +/-   ##
==========================================
+ Coverage   69.97%   70.38%   +0.41%     
==========================================
  Files          27       27              
  Lines        2348     2347       -1     
  Branches      383      383              
==========================================
+ Hits         1643     1652       +9     
+ Misses        632      620      -12     
- Partials       73       75       +2
```


| [Impacted Files](https://codecov.io/gl/pycqa/flake8/pull/227?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [src/flake8/checker.py](https://codecov.io/gl/pycqa/flake8/pull/227/diff?src=pr&el=tree#diff-c3JjL2ZsYWtlOC9jaGVja2VyLnB5) | `47.83% <100%> (+2.6%)` | :arrow_up: |
| [src/flake8/processor.py](https://codecov.io/gl/pycqa/flake8/pull/227/diff?src=pr&el=tree#diff-c3JjL2ZsYWtlOC9wcm9jZXNzb3IucHk=) | `64.87% <0%> (+0.41%)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gl/pycqa/flake8/pull/227?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gl/pycqa/flake8/pull/227?src=pr&el=footer). Last update [f834499...59218ca](https://codecov.io/gl/pycqa/flake8/pull/227?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Apr 15, 2018, 12:33

Can you describe the new behaviour in this case? I see that the `FileProcessor` raises (or doesn't handle) an `IOError` but what does Flake8 do with that?

**Comment by asottile** [comment]
In GitLab by @shukron on Apr 15, 2018, 13:35

`flake8` will catch that `IOError` and report it as an `E902`.

The following code is from the `FileChecker` class, which calls this method in its `__init__`.

```python
def _make_processor(self):
    try:
        return processor.FileProcessor(self.filename, self.options)
    except IOError:
        # If we can not read the file due to an IOError (e.g., the file
        # does not exist or we do not have the permissions to open it)
        # then we need to format that exception for the user.
        # NOTE(sigmavirus24): Historically, pep8 has always reported this
        # as an E902. We probably *want* a better error code for this
        # going forward.
        (exc_type, exception) = sys.exc_info()[:2]
        message = '{0}: {1}'.format(exc_type.__name__, exception)
        self.report('E902', 0, 0, message)
        return None
```

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Apr 15, 2018, 16:33

Ah, perfect. I forgot we had that.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Apr 15, 2018, 16:33

merged

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Apr 15, 2018, 16:33

mentioned in commit 52210149474a876c06922ae2c296445af5bdb108
